kinesis/src/main/scala/org/apache/pekko/stream/connectors/kinesis/impl/KinesisSchedulerSourceStage.scala

```diff
@@ -81,8 +81,14 @@ private[kinesis] class KinesisSchedulerSourceStage(
         override def shardRecordProcessor(): ShardRecordProcessor =
           new ShardProcessor(newRecordCallback)
       })
+      //Run the scheduler loop in a separate thread
+      val thread = new Thread(() => {
+        val result = Try {scheduler.run()}
+        callback.invoke(SchedulerShutdown(result))
+      }, s"KinesisSchedulerSource")
+      thread.setDaemon(true)
+      thread.start()
       schedulerOpt = Some(scheduler)
-      Future(scheduler.run()).onComplete(result => callback.invoke(SchedulerShutdown(result)))
```

Comment:
There is a `blocking` method on Scala's `Future` which might be a quick fix/solution?

Should be something like

```scala
Future {
  blocking {
    scheduler.run()
  }
}.onComplete(result => callback.invoke(SchedulerShutdown(result)))
```

Reply:
@jtjeferreira I'm not dead set against using a short lived daemon thread like you have in the current version of the PR but did you check the Future blocking approach?

Reply:
> using a short lived daemon thread

AFAICT this is will not be a short lived thread, but a long lived thread that will keep running as long as we want to read records from Kinesis.

That said, I don't think running this "loop" in the default dispatcher even with "blocking" is a good idea, as it would block a thread for a long time...

Reply:
> That said, I don't think running this "loop" in the default dispatcher even with "blocking" is a good idea, as it would block a thread for a long time...

What "loop" are you talking about? And by blocking a thread is the core point you are making is that it should be asynchronous rather than waiting for the `scheduler.run` operation to complete?

I am under the impression that the entire point of `Future` is it makes the enclosing operation asynchronous and what `blocking` does is it makes that asynchronous operation run on a pool that is designed to handle blocking operations (i.e. usually a CachedThreadPool of some kind however the way that `blocking` works is its up to the `ExecutionContext` that is pulled in to determine how to best handle blocking operations).

Reply:
> What "loop" are you talking about? 

I mean this loop 
https://github.com/awslabs/amazon-kinesis-client/blob/a1731dc49b2d76b8dd33781d5cabc5264b7079f9/amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java#L323-L325